gui/src-tauri/src/main.rs

```diff
@@ -111,13 +140,23 @@ fn main() {
 	let stdout_recv = if let Some(p) = run_path {
 		log::info!("Server found on path: {}", p.to_str().unwrap());
 
-		let java_folder = p.join("jre");
-		let (recv, _child) =
-			Command::new(java_folder.join("bin/java").to_str().unwrap())
-				.current_dir(p)
-				.args(["-Xmx512M", "-jar", "slimevr.jar", "--no-gui"])
-				.spawn()
-				.expect("Unable to start the server jar");
+		// Check if any Java already installed is compatible
+		let java_paths = valid_java_paths();
+		let jre = p.join("jre/bin/java");
+		let java_bin = jre
+			.exists()
+			.then(|| jre.to_string_lossy())
+			.or_else(|| java_paths.first().map(|x| x.0.to_string_lossy()));
```

Comment:
`let java_paths = valid_java_paths();` should be retrieved after checking the bundled Java version, as launching a bunch of child processes only to not use the result is kinda wasteful.

Reply:
ehhhhh, I know. But java_bin is a reference ``Cow<str>``.
What do you suggest